# 1. Inspección Inicial del Dataset
En esta sección realizamos la carga del dataset original en su formato crudo y una primera exploración de su estructura, dimensiones y tipos de datos para reunir evidencia inicial.

In [2]:
import pandas as pd

df = pd.read_json("../data/raw/streaming_users_dirty.json")

df.head()

,user_id,age,subscription_plan,monthly_watch_time_mins,country,favorite_genre,last_login_date,customer_support_tickets
0,10000,39,Estándar,805.8,Brasil,Crime,2025-03-04,99
1,10001,37,Estándar,1173.4,Colombia,Crime,2019-04-02,2
2,10002,28,Básico,401.0,Colombia,Crime,2018-04-13,0
3,10003,43,Básico,62.4,Uruguay,Thriller,2021-01-31,0
4,10004,51,Básico,477.8,Perú,Thriller,2020-09-30,1



Verificamos la cantidad total de registros (filas) y variables (columnas) que componen el conjunto de datos.

In [2]:
df.shape

(8160, 8)


Analizamos los tipos de datos asignados a cada columna y verificamos la consistencia general de la estructura.

In [3]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 8160 entries, 0 to 8159
Data columns (total 8 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   user_id                   8160 non-null   int64  
 1   age                       8160 non-null   int64  
 2   subscription_plan         8160 non-null   str    
 3   monthly_watch_time_mins   7967 non-null   float64
 4   country                   8160 non-null   str    
 5   favorite_genre            7920 non-null   str    
 6   last_login_date           7840 non-null   str    
 7   customer_support_tickets  8160 non-null   int64  
dtypes: float64(1), int64(3), str(4)
memory usage: 756.7 KB



Contabilizamos la cantidad de valores nulos presentes en cada una de las variables para orientar la posterior etapa de calidad y limpieza.

In [4]:
df.isnull().sum()

user_id                       0
age                           0
subscription_plan             0
monthly_watch_time_mins     193
country                       0
favorite_genre              240
last_login_date             320
customer_support_tickets      0
dtype: int64


Verificamos si existen filas idénticas repetidas dentro del dataset.

In [5]:
df.duplicated().sum()

np.int64(126)

In [4]:
df.describe()

,user_id,age,monthly_watch_time_mins,customer_support_tickets
count,8160.000000,8160.000000,7967.000000,8160.000000
mean,13995.433824,34.096814,1107.346153,1.800980
std,2310.810660,14.511304,5310.442604,11.334969
min,10000.000000,-5.000000,-120.000000,-1.000000
25%,11987.750000,25.000000,489.200000,0.000000
50%,13998.500000,33.000000,757.400000,1.000000
75%,15997.250000,42.000000,1045.700000,1.000000
max,17999.000000,150.000000,99999.000000,150.000000


utilizamos describe para ver si hay inconsistencias en los datos 

## Conclusiones de la Inspección Inicial

A partir de la exploración realizada, se determinó que el dataset está compuesto por 8.160 registros y 8 variables, que incluyen atributos numéricos y categóricos relacionados con usuarios de una plataforma de streaming.

Se detectaron las siguientes particularidades:

Registros duplicados: Se identificaron 126 filas completamente duplicadas, las cuales deberán ser eliminadas en la etapa de limpieza para evitar sesgos en los análisis posteriores.
Valores faltantes: Se encontraron valores nulos en las columnas monthly_watch_time_mins, favorite_genre y last_login_date, por lo que será necesario definir estrategias de imputación o eliminación según el tipo de variable.
Calidad de los datos: La presencia de duplicados y valores faltantes evidencia la necesidad de aplicar procesos de limpieza y tratamiento de datos antes de realizar análisis exploratorios o modelos de minería de datos, ademas encontramos inconsistencias como por ejemplo en la edad donde aparecen como -5 o 150 añoas.

Con este diagnóstico, el dataset queda preparado para iniciar la etapa de calidad y limpieza en 02_calidad_y_limpieza.ipynb.